streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -685,6 +685,47 @@ <VR> KStream<K, VR> flatMapValues(final ValueMapper<? super V, ? extends Iterabl
     <VR> KStream<K, VR> flatMapValues(final ValueMapperWithKey<? super K, ? super V, ? extends Iterable<? extends VR>> mapper,
                                       final Named named);
 
+    /**
+     * Marking the {@code KStream} as partitioned signals the stream is partitioned as intended,
+     * and does not require further repartitioning in downstream key changing operations.
+     *
+     * <p><em>
+     *     Note that {@link KStream#markAsPartitioned()} SHOULD NOT be used with interactive query(IQ) or {@link KStream#join}.
+     *     For reasons that when repartitions happen, records are physically shuffled by a composite key defined in the stateful operation.
+     *     However, if the repartitions were cancelled, records stayed in their original partition by its original key. IQ or joins
+     *     assumes and uses the composite key instead of the original key.
+     * </p></em>
```

Comment:
no need to add `</p>` -- JavaDocs `<p>` tag is a JavaDoc tag, not a HTML tag and does not need closing. (similar below)

Reply:
👍  wasn't sure as I saw the other javadocs doing it

Reply:
Resolved in the new PR.